compilation/platforms/solc.go

```diff
@@ -104,6 +104,18 @@ func (s *SolcCompilationConfig) Compile() ([]types.Compilation, string, error) {
 
 	// Create a compilation unit out of this.
 	compilation := types.NewCompilation()
+	if sourceList, ok := results["sourceList"]; ok {
+		if sourceListCasted, ok := sourceList.([]any); ok {
+			compilation.SourceList = make([]string, len(sourceListCasted))
+			for i := 0; i < len(sourceListCasted); i++ {
+				compilation.SourceList[i] = sourceListCasted[i].(string)
+			}
+		} else {
+			return nil, "", fmt.Errorf("could not parse compiled source artifact because 'sourcesList' was not a []string type")
```

Comment:
Maybe we could make this error message a bit more informative? hahaha

Perhaps we should say something like "incorrect JSON type for `sourcesList`, was expecting a []string"

Reply:
Agreed, though this applies in a few areas and all the error handling in these files does this. We nuked `truffle` support in this PR, and we might collapse `solc` support and just have one config that's crytic-compile based moving forward, until we re-evaluate in the future, so maybe it's best I'll leave this one for another time.

Anish's error handling PR might be a good one to touch this up in. I'll reference it so we don't lose track of it there (#107)